Czech translation

lobodpav opened the pull request:
Hello, the DFRobot guys.

I have translated the Maqueen robot into Czech, for my son who will find the robot under the Christmas tree and I thought the community might appreciate the translation as well 😊 

It was a surprisingly difficult task to do the translation right including method parameters.

I know I ask a lot, but I would appreciate your feedback, @almasy.

Below, find screenshots of all the translations. I thought it might make much more sense to visualise the translations for the review.

# Maqueen v4 JS blocks
<img width="342" alt="Screenshot 2024-11-24 at 11 28 56" src="https://github.com/user-attachments/assets/8c9acbeb-331f-43f0-876b-e5ca67e3eef2">
<img width="338" alt="Screenshot 2024-11-24 at 11 29 05" src="https://github.com/user-attachments/assets/19219362-fe25-4942-a8be-3bdb139aa988">

# Maqueen v5 JS blocks
<img width="342" alt="Screenshot 2024-11-24 at 11 29 18" src="https://github.com/user-attachments/assets/00b7456c-e57f-4b45-b471-bf9f50fa186f">
<img width="342" alt="Screenshot 2024-11-24 at 11 29 35" src="https://github.com/user-attachments/assets/d72b74d0-a041-439f-8b9a-5930486f7e0b">
<img width="342" alt="Screenshot 2024-11-24 at 11 29 45" src="https://github.com/user-attachments/assets/b737f196-66b4-4638-b4b0-a18e9b6ab2dc">
<img width="340" alt="Screenshot 2024-11-24 at 11 29 56" src="https://github.com/user-attachments/assets/fd52ba0a-6653-40c7-95d3-c92903e4c18a">
<img width="342" alt="Screenshot 2024-11-24 at 11 40 00" src="https://github.com/user-attachments/assets/3cdd2584-9daa-4480-ad11-85991551de40">
<img width="344" alt="Screenshot 2024-11-24 at 11 40 12" src="https://github.com/user-attachments/assets/8799d5c3-edda-47af-bac7-e95cdabb8d5e">

# Maqueen v4 and v5 blocks
<img width="1071" alt="Screenshot 2024-11-24 at 11 28 06" src="https://github.com/user-attachments/assets/421d02bd-d14e-4891-8c91-e6eab5cc5b28">
<img width="659" alt="Screenshot 2024-11-24 at 11 28 22" src="https://github.com/user-attachments/assets/7ef655fc-bf9a-4661-8248-fb4965b89e42">
<img width="799" alt="Screenshot 2024-11-24 at 11 28 34" src="https://github.com/user-attachments/assets/0fe28fe9-b230-402d-a0c0-8f15fcedb339">

lobodpav committed:
Czech translation

almasy commented on _locales/cs/maqueen-jsdoc-strings.json:
```diff
@@ -0,0 +1,79 @@
+  "maqueen.readPatrol": "Načte digitální hodnotu ze senzoru čáry (1 = čára přítomna, 0 = čára není přítomna).",
+  "maqueen.readPatrol|param|patrol": "senzor čáry, např. Patrol.PatrolLeft",
+  "maqueen.servoRun": "Nastaví úhel natočení servomotoru.",
+  "maqueen.writeLED": "Zapne, nebo vypne ybranou LED.",
```
There's a typo (missing letter 'v') here. You meant to write "Zapne, nebo vypne vybranou LED."

almasy commented on _locales/cs/maqueen-jsdoc-strings.json:
```diff
@@ -0,0 +1,79 @@
+  "maqueen.motorRun|param|speed": "rychlost otáčení, např. 100",
+  "maqueen.motorStop": "Zastaví vybraný motor.",
+  "maqueen.motorStop|param|motors": "motor, který chceš zastavit",
+  "maqueen.readPatrol": "Načte digitální hodnotu ze senzoru čáry (1 = čára přítomna, 0 = čára není přítomna).",
```
I'd rather suggest to use "logickou" instead of "digitální". All numbers micro:bit internally works with are digital (i.e. non-analog). However, what the term "digitální" represents in Czech is much broader than what is meant in this context. It's even indicated by Voltage.High and Voltage.Low enums. 

This will also make your translation consistent with MakeCode translating "read digital" to "logická hodnota z pinu".

almasy commented on _locales/cs/maqueen-jsdoc-strings.json:
```diff
@@ -0,0 +1,79 @@
+  "maqueen.Ultrasonic": "Vrátí vzdálenost robota od překážky v cm.",
+  "maqueen.ltEvent": "Spustí obsah bloku, pokud zvolený senzor čáry najde nebo nenajde čáru.",
+  "maqueen.ltEvent|param|value": "senzor čáry, např. Patrol1.PatrolLeft",
+  "maqueen.ltEvent|param|vi": "digitální hodnota senzoru čáry, např. Voltage.High",
```
Consider using "logická" instead of "digitální". Reasoning is the same as for maqueen.readPatrol

almasy commented on _locales/cs/maqueen-jsdoc-strings.json:
```diff
@@ -0,0 +1,79 @@
+  "Maqueen_V5.patrolling": "Zapne nebo vypne sledování čáry.",
+  "Maqueen_V5.patrolling|param|patrol": "stav sledování čáry, např. Patrolling.ON.",
+  "Maqueen_V5.readLightIntensity": "Načte intenzitu levé, nebo pravé LED.",
+  "Maqueen_V5.readPatrol": "Načte digitální hodnotu ze senzoru čáry (1 = čára přítomna, 0 = čára není přítomna).",
```
Consider using "logickou" instead of "digitální". Reasoning is the same as for maqueen.readPatrol

almasy commented on _locales/cs/maqueen-jsdoc-strings.json:
```diff
@@ -0,0 +1,79 @@
+  "Maqueen_V5.patrolling|param|patrol": "stav sledování čáry, např. Patrolling.ON.",
+  "Maqueen_V5.readLightIntensity": "Načte intenzitu levé, nebo pravé LED.",
+  "Maqueen_V5.readPatrol": "Načte digitální hodnotu ze senzoru čáry (1 = čára přítomna, 0 = čára není přítomna).",
+  "Maqueen_V5.readPatrolData": "Načte analogovou hodnotu ze senzoru čáry v rozsahu od -128 do 127.",
```
Consider "číselnou" instead of "analogovou". Again, technically the value micro:bit receives is digital (because of ADC on the PIN). But what matters is that the returned value isn't just "boolean on/off" but an actual number.

Last, but not least, there's also the consistency with the MakeCode translation ("analog read pin" -> "číslo z pinu")

almasy commented on _locales/cs/maqueen-jsdoc-strings.json:
```diff
@@ -0,0 +1,79 @@
+  "Maqueen_V5.motorStop|param|motors": "motor k zastavení",
+  "Maqueen_V5.patrolSpeed": "Nastaví rychlost sledování čáry.",
+  "Maqueen_V5.patrolSpeed|param|speed": "rychlost sledování čáry",
+  "Maqueen_V5.patrolling": "Zapne nebo vypne sledování čáry.",
```
Nitpicking comment - you forgot the coma before "nebo". i.e. "Zapne, nebo vypne sledování čáry."

almasy commented on _locales/cs/maqueen-jsdoc-strings.json:
```diff
@@ -0,0 +1,79 @@
+  "Maqueen_V5.bottomShowColor|param|rgb": "RGB barva, např. NeoPixelColors.Red",
+  "Maqueen_V5.colors": "Získá numerickou RGB hodnotu zvolené barvy.",
+  "Maqueen_V5.colors|param|color": "zvolená barva, např. NeoPixelColors.Red",
+  "Maqueen_V5.getBatteryData": "Načte úroveň nabití baterie v rozsahu 0-100%",
```
I may be nitpicking here again, but I don't think "úroveň nabití" is term which should be used in context of alkaline batteries (AFAIK these (primární články) are NOT designed to be reusable. Yes, some can be recharged to a certain degree, but it can be a fire hazard). 

I know it's not precise either, but I'd rather use "Načte zbývající kapacitu baterie v rozsahu 0-100%".

lobodpav replied:
Hmm, that one was difficult. I was playing with words for a while and no translation seemed to be good enough 😏 

Your translation seems better than mine, though. Will use it.

almasy commented on _locales/cs/maqueen-jsdoc-strings.json:
Seems like the folks at DFRobot have decided to comment out all of the BLE code. Therefore I'm not looking at the BLE related translations.

almasy commented on _locales/cs/maqueen-strings.json:
Seems like the folks at DFRobot have decided to comment out all of the BLE code. Therefore I'm not looking at the BLE related translations.

almasy commented on maqueen.ts:
Not looking into this file due to changes done by DFRobot people...

almasy commented on _locales/cs/maqueen-strings.json:
```diff
@@ -0,0 +1,128 @@
+  "maqueen.motorRun|block": "zapni %index(levý motor) | směrem %Dir(vpřed)| rychlostí %speed(100)",
+  "maqueen.motorStop|block": "zastav %motors(levý motor)",
+  "maqueen.readPatrol|block": "stav %patrol(levého) senzoru čáry",
+  "maqueen.servoRun|block": "otoč servomotor |%index(S1) |na |%angle(90) stupňů",
```
Again, this might be bit nitpicky comment. However, "natoč servomotor S1 na 90 stupňů" sounds more natural to me. I'd use "otoč servomotor" if I wanted to express the difference to the current state (i.e. "otoč servomotor o 90 stupňů" means rotate 90 degrees more from the current position).

almasy commented on _locales/cs/maqueen-strings.json:
```diff
@@ -0,0 +1,128 @@
+  "Maqueen_V5.readPatrolData|block": "analogová data z %Patrol(levého) senzoru čáry",
+  "Maqueen_V5.readPatrol|block": "stav %Patrol(levého) senzoru čáry",
+  "Maqueen_V5.readVersion|block": "verze robota",
+  "Maqueen_V5.servoRun|block": "otoč servomotor |%index(S0) |na |%angle(90) stupňů",
```
Consider using "natoč servomotor". See my reasoning at maqueen.servoRun

almasy commented on _locales/cs/maqueen-strings.json:
```diff
@@ -0,0 +1,128 @@
+  "Maqueen_V5.patrolSpeed|block": "nastav rychlost sledování čáry na |%PatrolSpeed(1)",
+  "Maqueen_V5.patrolling|block": "%Patrolling(zapni) sledování čáry",
+  "Maqueen_V5.readLightIntensity|block": "intenzita LED světla %type(vlevo)",
+  "Maqueen_V5.readPatrolData|block": "analogová data z %Patrol(levého) senzoru čáry",
```
Consider using "číselná hodnota z %Patrol(levého) senzoru čáry". Please see my comment to this topic in "maqueen-jsdoc-strings.json".

almasy commented on _locales/cs/maqueen-strings.json:
```diff
@@ -0,0 +1,128 @@
+  "Maqueen_V5.bottomSetColor|block": "nastav barvu RGB LED |na pozici %index |na %rgb=neopixel_colors(červenou)",
+  "Maqueen_V5.bottomShowColor|block": "nastav barvu všech RGB LED |na %rgb=neopixel_colors(červenou)",
+  "Maqueen_V5.colors|block": "%color(červená)",
+  "Maqueen_V5.getBatteryData|block": "úroveň nabití %type(alkalické) baterie",
```
Consider using "zbývající kapacita %type(alkalické) baterie". Please see my comment to this topic in "maqueen-jsdoc-strings.json".

almasy commented on _locales/cs/maqueen-strings.json:
```diff
@@ -0,0 +1,128 @@
+  "Maqueen_V5.SpeedGrade.speed5|block": "5",
+  "Maqueen_V5.Ultrasonic|block": "vzdálenost z ultrazvukového senzoru v cm",
+  "Maqueen_V5.Voltage.High|block": "vysoké",
+  "Maqueen_V5.Voltage.Low|block": "nízké",
```
I believe this will be used in the same way as maqueen.Voltage.Low, hence this translation is not correct. The question aimed at the DFRobot team is - why there's no V5 "on line tracking sensor" block available?

lobodpav replied:
I did not notice the enum was not used within the code at all. Good catch.

almasy commented on _locales/cs/maqueen-strings.json:
```diff
@@ -0,0 +1,128 @@
+  "Maqueen_V5.SpeedGrade.speed4|block": "4",
+  "Maqueen_V5.SpeedGrade.speed5|block": "5",
+  "Maqueen_V5.Ultrasonic|block": "vzdálenost z ultrazvukového senzoru v cm",
+  "Maqueen_V5.Voltage.High|block": "vysoké",
```
see "Maqueen_V5.Voltage.Low|block"

almasy commented on _locales/cs/maqueen-strings.json:
```diff
@@ -0,0 +1,128 @@
+  "Maqueen_V5.NeoPixelColors.Green|block": "zelenou",
+  "Maqueen_V5.NeoPixelColors.Indigo|block": "indigo",
+  "Maqueen_V5.NeoPixelColors.Orange|block": "oranžovou",
+  "Maqueen_V5.NeoPixelColors.Purple|block": "fialovou",
```
I guess you accidentally swapped translations for Maqueen_V5.NeoPixelColors.Purple and Maqueen_V5.NeoPixelColors.Violet.

Violet = fialová (the rainbow colour)
Purple = purpurová OR (less common) nachová (as the redish/blueish/pinkish colour of clothes important / powerful people liked to wear in order to distinguish themselves).

almasy commented on _locales/cs/maqueen-strings.json:
```diff
@@ -0,0 +1,128 @@
+  "Maqueen_V5.NeoPixelColors.Orange|block": "oranžovou",
+  "Maqueen_V5.NeoPixelColors.Purple|block": "fialovou",
+  "Maqueen_V5.NeoPixelColors.Red|block": "červenou",
+  "Maqueen_V5.NeoPixelColors.Violet|block": "purpurovou",
```
I guess you accidentally swapped translations for Maqueen_V5.NeoPixelColors.Purple and Maqueen_V5.NeoPixelColors.Violet.

Violet = fialová (the rainbow colour)
Purple = purpurová OR (less common) nachová (as the redish/blueish/pinkish colour of clothes important / powerful people liked to wear in order to distinguish themselves).

almasy commented on _locales/cs/maqueen-strings.json:
```diff
@@ -0,0 +1,128 @@
+  "Maqueen_V5.CarLightColors.Blue|block": "modrou",
+  "Maqueen_V5.CarLightColors.Cyan|block": "azurovou",
+  "Maqueen_V5.CarLightColors.Green|block": "zelenou",
+  "Maqueen_V5.CarLightColors.Purple|block": "fialovou",
```
I don't have the maqueen v5 car to judge the actual colour, but purple is usually translated as purpurová. The "fialová" represents a different colour.

Considering the context where the NeoPixelColors.Purple defines it as red and blue mix (0xFF00FF), I'd say purpurová is proper translation.

almasy reviewed (changes requested):
Hi Pavel!

I've finally managed to have a look at your translations. In general, I like it. Well done. I've added few (mostly nitpicking) comments / improvement suggestions.

P.S.
Seems like DFRobot team doesn't want to include the BLE part. I'd suggest you drop those changes for now and focus on covering the officially released functionality.

lobodpav committed:
Code review fixes (Czech translation)

lobodpav commented:
> Hi Pavel!
> 
> I've finally managed to have a look at your translations. In general, I like it. Well done. I've added few (mostly nitpicking) comments / improvement suggestions.
> 
> P.S. Seems like DFRobot team doesn't want to include the BLE part. I'd suggest you drop those changes for now and focus on covering the officially released functionality.

Hey Peter.

Thanks for your thorough review! Translating so much stuff was incredibly difficult and I was quite losing my mind amongst all the words 😊 Having another pair of eyes on this PR was much appreciated.

I have incorporated all your suggestions and also removed my BLE changes. Not sure why they removed it during this PR waiting for the review. A pity.

lobodpav commented:
DFRobot guys, could you have a look? (@tangjie133 perhaps?).

LinDong88 reviewed (approved):
Thank you very much for conducting a comprehensive review of the translation and adding the Czech language version. Welcome to continue raising your questions. We will do our best to solve the problems you encounter in the future. Best wishes!

LinDong88 committed:
Merge branch 'master' into master
